### PR TITLE
Refactored PR #949 and edited the comments/docstrings

### PR DESCRIPTION
**Pull request by erezsh**
Also now only testing on LALR, to prevent needless compute and save the planet.

For reference (because removed from code):
```
REPEAT_BREAK_THRESHOLD:

# For a grammar  of the form start: "A"~0..N, these are the timing stats:
#  N  t
# 10 0.000
# 20 0.004
# 30 0.016
# 40 0.049
# 50 0.109
# 60 0.215
# 70 0.383
# 80 0.631
```

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -2226,6 +2226,35 @@ def test_ranged_repeat_terms(self):
             self.assertRaises((ParseError, UnexpectedInput), l.parse, u'ABB')
             self.assertRaises((ParseError, UnexpectedInput), l.parse, u'AAAABB')
 
+        @unittest.skipIf(PARSER != 'lalr', "We only need to test rule generation, we know BNF is solid on all parsers")
```
Comment (MegaIng) [inline tests/test_parser.py:2229]: Then just move it into `test_grammar`

File: tests/test_parser.py [hunk 2]
```diff
@@ -2226,6 +2226,35 @@ def test_ranged_repeat_terms(self):
...
+        def test_ranged_repeat_large(self):
+            # Large is currently arbitrarily chosen to be large than 20
```
Comment (MegaIng) [inline tests/test_parser.py:2231]: This is wrong. We also might want to dynamically change the 60 below.

**Review by MegaIng — CHANGES_REQUESTED** [review]
Were the tests really a performance problem for you? For me they ran a lot faster than as example the regex test.

**Comment by MegaIng** [comment]
BTW, if you want to run any of the performance tests for yourself, I uploaded my scripts into [a repo](https://github.com/MegaIng/lark-performance-tests)

File: tests/test_parser.py (hunk 1 above)
Comment (erezsh) [reply tests/test_parser.py]: Good idea!

**Review by MegaIng — APPROVED** [review]

**Comment by erezsh** [comment]
> Were the tests really a performance problem for you? For me they ran a lot faster than as example the regex test.

Well, it added almost 10 seconds, and now it takes 0.3 secs. It wasn't critical to remove, but I also think we should generally avoid long computations when they don't test anything.

**Comment by erezsh** [comment]
Btw, what does you think about the coverage report? Useful or annoying?

**Comment by MegaIng** [comment]
In general good, but I am having a hard time interpreting them sometimes.

**Pull request merged**